src/omv/common/sensor_utils.c

```diff
@@ -524,7 +524,7 @@ __weak int sensor_set_pixformat(pixformat_t pixformat)
     // fit in RAM it will just be switched back again to BAYER, so we keep the current format unchanged.
     uint32_t size = framebuffer_get_buffer_size();
     if ((sensor.pixformat == PIXFORMAT_BAYER)
-            && (pixformat == PIXFORMAT_RGB565)
+            && ((pixformat == PIXFORMAT_RGB565) || (pixformat == PIXFORMAT_YUV422))
```

Comment:
```suggestion
            && (pixformat->bpp == 2)
```
Seems we only care about 2 bpp here, also please fix the comment above.

Reply:
True

Reply:
Not possible to fix as PIXFORMAT is not a struct. You implemented it as an ENUM so you can't get the bpp without shifting and masking. If I do this it makes the code brittle.

Reply:
If you'd like to update the pixformat enum to be a struct so this can be done then this can be fixed. The check should be for is_color and bpp == 2.

Reply:
No you're right it's just an enum, not important to fix now.